pysidt/sidt.py

```diff
@@ -866,21 +891,36 @@ def estimate_uncertainty(self):
             self.node_uncertainties.update(
                 {node.name: 1.0 for i, node in enumerate(nodes)}
             )
+        
+        for node in self.nodes.values():
+            n = node.rule.num_data
+            if n > 1:
+                t = scipy.stats.t.ppf((1 + confidence_level) / 2, n - 1)
+                node.rule.uncertainty = t * self.node_uncertainties[node.name] / np.sqrt(n)
+        
+        for node in self.nodes.values():
+            n = node.rule.num_data
+            if n == 1:
+                node.rule.uncertainty = node.parent.rule.uncertainty
+
 
     def assign_depths(self):
         root = self.root
         _assign_depths(root)
 
-    def evaluate(self, mol):
+    def evaluate(self, mol, estimate_uncertainty=True):
```

Comment:
default estimate_uncertainty=False for now

Reply:
Done